File: Java/LifecycleDemo/app/src/main/java/com/example/lifecycledemo/MainActivityViewModel.java
package com.example.lifecycledemo;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MainActivityViewModel extends ViewModel {
    private int clickCount = 0;
    private MutableLiveData<Integer> countLiveData = new MutableLiveData<>();

    MutableLiveData<Integer> getInitialCount() {
        countLiveData.setValue(clickCount);
        return countLiveData;
    }

    void updateCurrentCount() {
        ++clickCount;
        countLiveData.setValue(clickCount);
    }
}

File: Kotlin/LiveDataDemo/app/src/main/java/com/example/livedatademo/MainActivityViewModel.kt
package com.example.livedatademo

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MainActivityViewModel(starterTotal: Int) : ViewModel() {
    private var total = MutableLiveData<Int>()
    val totalData: LiveData<Int> get() = total

    init {
        total.value = starterTotal
    }

    fun setTotal(input: Int) {
        total.value = (total.value)?.plus(input)
    }
}